### DISCUSSION TITLE
How to enrich data with external HTTP-request?

### DISCUSSION THREAD

**Discussion by D13410N3** [vectordotdev/vector#14997] [General]
Hello! I'd like to enrich data with response from external URL (just like http-module in logstash) usinf VRL. So, I'm making HTTP-request with some existing fields and taking some fields from response.

I didn't find any function for this in docs. Is it possible?

**Comment by jszwedko** [comment]
Hi @D13410N3 ! This isn't currently possible natively in Vector (it is being tracked by https://github.com/vectordotdev/vector/issues/1385) but can be achieved by using the `lua` transform which gives you access to the full Lua language.

**Comment by D13410N3** [comment]
@jszwedko can you tell what modules are included in default engine for http-requests & json parsing? I can't require any module like http.request & json in configuration

**Reply by jszwedko** [reply]
Hi @D13410N3 . I think you'll need to download and install lua modules for both HTTP requests and JSON parsing. I don't believe facilities for those are included in the language itself.
